src/recipe/Baked.Recipe.Service.Application/CodeGeneration/CodeGenerationLayer.cs

```diff
@@ -1,46 +1,107 @@
+    readonly string _location = default!;
+
+    public CodeGenerationLayer()
+    {
+        _location = Path.GetDirectoryName(Assembly.GetEntryAssembly()?.Location) ??
+            throw new("'EntryAssembly' should have existed with valid location");
+        _location = Path.Combine(_location, Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") ?? "Development");
+    }
 
     protected override PhaseContext GetContext(GenerateCode phase) =>
         phase.CreateContext(_generatedAssemblies);
 
+    protected override PhaseContext GetContext(Compile phase) =>
+        phase.CreateContext(_generatedFiles);
+
     protected override IEnumerable<IPhase> GetPhases()
     {
-        yield return new GenerateCode(_generatedAssemblies);
-        yield return new Compile();
+        yield return new LoadGeneratedAssemblies(_location);
+    }
+
+    protected override IEnumerable<IPhase> GetBakePhases()
+    {
+        yield return new GenerateCode(_location, _generatedAssemblies, _generatedFiles);
+        yield return new Compile(_location);
+        yield return new CreateFiles(_location);
     }
 
-    public class GenerateCode(IGeneratedAssemblyCollection _generatedAssemblies)
+    public class GenerateCode(string _location, IGeneratedAssemblyCollection _generatedAssemblies, IGeneratedFileCollection _generatedFiles)
         : PhaseBase<DomainModel>(PhaseOrder.Early)
     {
         protected override void Initialize(DomainModel _)
         {
+            if (Directory.Exists(_location))
+            {
+                Directory.Delete(_location, true);
+            }
+
+            Directory.CreateDirectory(_location);
+
             Context.Add(_generatedAssemblies);
+            Context.Add(_generatedFiles);
         }
     }
 
-    public class Compile()
-        : PhaseBase<IGeneratedAssemblyCollection>(PhaseOrder.Latest)
+    public class Compile(string _location)
+        : PhaseBase<IGeneratedAssemblyCollection>(PhaseOrder.Late)
     {
         protected override void Initialize(IGeneratedAssemblyCollection generatedAssemblies)
         {
-            var provider = new GeneratedAssemblyProvider();
             foreach (var descriptor in generatedAssemblies)
             {
-                var assembly = new Compiler(descriptor).Compile();
+                new Compiler(descriptor).Compile(_location, $"Baked.g.{descriptor.Name}");
+            }
+        }
+    }
+
+    public class CreateFiles(string _location)
+        : PhaseBase<IGeneratedFileCollection>(PhaseOrder.Latest)
+    {
+        protected override void Initialize(IGeneratedFileCollection generatedFiles)
+        {
+            foreach (var descriptor in generatedFiles)
+            {
+                using var file = new FileStream(Path.Combine(_location, $"{descriptor.Name}.{descriptor.Extension}"), FileMode.Create);
+                file.Write(Encoding.UTF8.GetBytes(descriptor.Content));
+            }
+        }
+    }
```

Comment:
this looks like it belongs to Generate phase, and on phase context dispose, it can write to file instead of having a separate phase just for file writing.